src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,119 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class RegistrationServiceTest {
+    private static RegistrationService registrationService;
+    private User newUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        newUser = new User();
+        newUser.setLogin("login");
+        newUser.setPassword("password");
+        newUser.setAge(18);
+    }
+
+    @Test
+    @Order(1)
+    void register_userLoginIsNull_notOk() {
+        newUser.setLogin(null);
+        assertThrows(NullPointerException.class, () -> registrationService.register(newUser));
+    }
+
+    @Test
+    @Order(2)
+    void register_userLoginLengthTooShort_notOk() {
+        newUser.setLogin("abc");
+        assertThrows(RuntimeException.class, () -> registrationService.register(newUser));
+    }
+
+    @Test
+    @Order(2)
+    void register_userLoginLengthTooLong_notOk() {
+        newUser.setLogin("abcdefghijklmnopqrstuvwxyzABCDE");
+        assertThrows(RuntimeException.class, () -> registrationService.register(newUser));
+    }
+
+    @Test
+    @Order(1)
+    void register_userPasswordIsNull_notOk() {
+        newUser.setPassword(null);
+        assertThrows(NullPointerException.class, () -> registrationService.register(newUser));
+    }
+
+    @Test
+    @Order(3)
+    void register_userPasswordIsTooShort_notOk() {
+        newUser.setPassword("abc");
```

Comment:
```suggestion
        newUser.setPassword("abcde");
```
better create notValid password length closer to acceptable(saw others were getting a recomendation to change)